db/migrate/20171121000000_prepare_embed_columns_in_preview_cards.rb

```diff
@@ -0,0 +1,31 @@
+require Rails.root.join('lib', 'mastodon', 'migration_helpers')
+
+# This migration will need servers running a revision equals to or later than
+# commit 067c539ffae4628b7153c6d57f511a92e4693765 for up direction.
+# For down direction, it will need a former revision.
+class PrepareEmbedColumnsInPreviewCards < ActiveRecord::Migration[5.1]
+  include Mastodon::MigrationHelpers
+
+  def up
+    safety_assured do
+      cleanup_concurrent_column_rename :preview_cards, :height, :embed_height
+      cleanup_concurrent_column_rename :preview_cards, :width, :embed_width
+      cleanup_concurrent_column_rename :preview_cards, :html, :embed_html
+    end
+
+    status_ids = Status.joins(:preview_cards)
```

Comment:
There is no need to join Status. I think that this query is enough.

```
PreviewCard.where(type: 'photo').joins('JOIN preview_cards_statuses ON preview_cards_statuses.preview_card_id = preview_cards.id').pluck(:status_id)
```

```
SELECT "status_id" FROM "preview_cards" JOIN preview_cards_statuses ON preview_cards_statuses.preview_card_id = preview_cards.id WHERE "preview_cards"."type" = 1
```

Reply:
You are right. However, I still prefer a query without any raw SQL code. It is not performance-critical and fine to have the little overhead.

Reply:
Ok, I see.